services/QuillLMS/client/app/bundles/PremiumHub/components/usage_snapshots/snapshotCount.tsx

```diff
@@ -111,16 +113,18 @@ const SnapshotCount = ({ label, size, queryKey, searchCount, selectedGrades, sel
     pusherChannel?.bind(PUSHER_EVENT_KEY, (body) => {
       const { message, } = body
 
-      const queryKeysAreEqual = message.query === queryKey
-      const timeframesAreEqual = message.timeframe === selectedTimeframe
-      const schoolIdsAreEqual = selectionsEqual(message.school_ids, selectedSchoolIds)
-      const teacherIdsAreEqual = selectionsEqual(message.teacher_ids, selectedTeacherIds)
-      const classroomIdsAreEqual = selectionsEqual(message.classroom_ids, selectedClassroomIds)
-      const gradesAreEqual =  selectionsEqual(message.grades, selectedGrades?.map(grade => String(grade))) || (!message.grades && !selectedGrades.length)
+      const filterTarget = [
+        queryKey,
+        selectedTimeframe,
+        selectedSchoolIds.join('-'),
+        selectedGrades?.join('-'),
+        selectedTeacherIds?.join('-'),
+        selectedClassroomIds?.join('-')
+      ].join('-')
 
-      if (queryKeysAreEqual && timeframesAreEqual && schoolIdsAreEqual && gradesAreEqual && teacherIdsAreEqual && classroomIdsAreEqual) {
-        getData()
-      }
+      const filterHash = md5(filterTarget)
```

Comment:
Can we extract this out into a function to make it testable. I'd like you to add a test for this.